shared/domain/src/commonMain/kotlin/network/bisq/mobile/data/service/network/ConnectivityService.kt

```
@@ -38,14 +68,68 @@ abstract class ConnectivityService :
+                }
+                if (_status.value == ConnectivityStatus.RECONNECTING) {
+                    return
+                }
+                cancelReconnectingTimeoutJob()
+                _status.value = ConnectivityStatus.RECONNECTING
+                var scheduledJob: Job? = null
+                scheduledJob =
+                    serviceScope.launch {
+                        try {
+                            delay(maxReconnectingDurationMs)
+                        } catch (e: CancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)